File: archive-site-frontend/src/app/models/activity.ts
import { Model } from 'src/app/models/model';
import { ActivityType } from 'src/app/models/activity-type';
import { User } from 'src/app/models/user';

export class Activity extends Model {
  constructor(
    id: number = 0,
    public UserId: number = 0,
    public Message: string = undefined,
    public EntityType: string = undefined,
    public EntityId: number = undefined,
    public Type: ActivityType = undefined,
    public CreatedTime: Date = undefined,
    public User?: User) {
    super(id);
  }
}

File: archive-site-backend/src/ArchiveSiteBackend.Data/Activity.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace ArchiveSite.Data {
    public class Activity : EntityBase<Activity> {
        [Required]
        public Int64 UserId { get; set; }
        [Required]
        public String Message { get; set; }

        public Int64? EntityId { get; set; }
        [MaxLength(100)]
        public String EntityType { get; set; }

        [Required]
        public ActivityType Type { get; set; }

        [Required]
        public DateTimeOffset CreatedTime { get; set; }

        public virtual User User { get; set; }
    }
}
